include/dataset.hpp

```diff
@@ -52,17 +54,55 @@ class CustomDataset : public torch::data::Dataset<CustomDataset> {
         labels = process_labels(list_labels);
         ds_size = states.size();
     };
-    
+
     torch::data::Example<> get(size_t index) override {
         /* This should return {torch::Tensor, torch::Tensor} */
         torch::Tensor sample_img = states.at(index);
         torch::Tensor sample_label = labels.at(index);
         return {sample_img.clone(), sample_label.clone()};
     };
-    
+
+    // Visualize batch of data (by default 3x3)
+    void show_batch(int batch_size = 3) {
+        cv::Mat* img_varray = new cv::Mat[batch_size*batch_size];
+        for(int i = 0; i < batch_size*batch_size; i++) {
+            torch::Tensor out_tensor = get(i).data.squeeze().detach().permute({1, 2, 0});
+            out_tensor = out_tensor.clamp(0, 255).to(torch::kCPU).to(torch::kU8);
+            *(img_varray + i) = cv::Mat::eye(out_tensor.sizes()[0], out_tensor.sizes()[1], CV_8UC3);
+            std::memcpy((img_varray + i)->data, out_tensor.data_ptr(), sizeof(torch::kU8) * out_tensor.numel());
+        }
+        cv::Mat out(256, 256, CV_8UC3);
+        cv::Mat temp_out(256, 256, CV_8UC3);
+        for(int vconcat_times = 0; vconcat_times < batch_size; vconcat_times++) {
+            cv::cvtColor(*(img_varray + vconcat_times*batch_size), out, cv::COLOR_BGR2RGB);
+            for(int hconcat_times = vconcat_times*batch_size; hconcat_times < (vconcat_times+1)*batch_size - 1; hconcat_times++) {
+                cv::cvtColor(*(img_varray + hconcat_times + 1), *(img_varray + hconcat_times + 1), cv::COLOR_BGR2RGB);
+                cv::hconcat(out, *(img_varray + hconcat_times + 1), out);
+            }
+            if(vconcat_times == 0)
+                temp_out = out;
+            else {
+                cv::vconcat(temp_out, out, temp_out);
+            }
+        }
+        cv::cvtColor(temp_out, temp_out, cv::COLOR_BGR2RGB);
+        cv::imwrite("out.jpg", temp_out);
+        std::cout << "Image saved as out.jpg" << std::endl;
+    }
+
+    // Visualizes sample at the given index
+    void show_sample(int index) {
+        cv::Mat sample_img(64, 64, CV_8UC3);
```

Comment:
Remove manual entry of image size to `cv::Mat::size()` value.

Reply:
Resolved here: https://github.com/BuffetCodes/DCGAN-CelebA-PyTorch-CPP/pull/3/commits/f54f104800e0ae5b0334b94bfe266a0c0b78d2ad